### PR TITLE
FIX: FetchMany(number of rows) ignores batch size with LOB columns

### PR DESCRIPTION
**Pull request by dlevy-msft-sql**
### Work Item / Issue Reference  

> GitHub Issue: #345 

-------------------------------------------------------------------
### Summary   
Fixes FetchMany(number of rows) ignores batch size when table contains an LOB

### CODE REVIEW

File: mssql_python/pybind/ddbc_bindings.cpp [hunk 1]
```diff
@@ -3870,22 +3870,23 @@ SQLRETURN FetchMany_wrap(SqlHandlePtr StatementHandle, py::list& rows, int fetch
         }
     }
 
+    SQLULEN numRowsFetched = 0;
```
Comment (sumitmsft) [inline mssql_python/pybind/ddbc_bindings.cpp:4038]: The `numRowsFetched` variable is used differently in the two code paths:

LOB path: Requires = 0 initialization (used as loop counter)
Non-LOB path: Initial value ignored (overwritten by `SQL_ATTR_ROWS_FETCHED_PTR`)

Adding a comment to clarify this dual purpose would be appropariate:
This will help future maintainers understand the design decision.


```C++
// Initialized to 0 for LOB path counter; overwritten by ODBC in non-LOB path
SQLULEN numRowsFetched = 0;

File: tests/test_004_cursor.py [hunk 2]
```diff
@@ -947,12 +973,32 @@ def test_fetchmany(cursor):
...
 def test_fetchall(cursor):
+    """Test fetching all rows"""
+    cursor.execute(
+        "SELECT id, bit_column, tinyint_column, smallint_column, bigint_column, integer_column, float_column, wvarchar_column, time_column, datetime_column, date_column, real_column FROM #pytest_all_data_types"
+    )
+    rows = cursor.fetchall()
+    assert isinstance(rows, list), "fetchall should return a list"
+    assert len(rows) == len(PARAM_TEST_DATA), "Incorrect number of rows returned"
+
+
```
Comment (sumitmsft) [inline tests/test_004_cursor.py:1212]: Add some additional test cases to cover edge cases:

1. Test fetchmany with size=0 for LOB columns
2. Test fetchmany requesting more rows than exist with LOB columns
3. Test fetchmany on empty result set with LOB columns
4. Test fetchmany with very large LOB column data (large_data = "x" * 10000)
